### PR TITLE
[DOCS] Update sharepoint-online connector perms

### PR DESCRIPTION
**Pull request by leemthompo**
Follow up on https://github.com/elastic/connectors/pull/2762

### CODE REVIEW

File: docs/reference/connector/docs/connectors-sharepoint-online.asciidoc [hunk 1]
```diff
@@ -87,13 +87,13 @@ Select an expiration date. (At this expiration date, you will need to generate a
...
 Graph API
-- Sites.Read.All
+- Sites.Selected
 - Files.Read.All
 - Group.Read.All
 - User.Read.All
 
 Sharepoint
-- Sites.Read.All
+- * Sites.Selected
```
Comment (leemthompo) [inline docs/reference/connector/docs/connectors-sharepoint-online.asciidoc:96]:
```suggestion
- Sites.Selected
```

File: docs/reference/connector/docs/connectors-sharepoint-online.asciidoc [hunk 2]
```diff
@@ -87,13 +87,13 @@ Select an expiration date. (At this expiration date, you will need to generate a
...
+- Sites.Selected
 - Files.Read.All
 - Group.Read.All
 - User.Read.All
 
 Sharepoint
-- Sites.Read.All
+- Sites.Selected
 ```
 * **Grant admin consent**, using the `Grant Admin Consent` link from the permissions screen.
```
Comment (leemthompo) [inline docs/reference/connector/docs/connectors-sharepoint-online.asciidoc:98]:
```suggestion
NOTE: If the `Comma-separated list of sites` configuration is set to `*` or if a user enables the toggle button `Enumerate all sites`, the connector requires `Sites.Read.All` permission.

* **Grant admin consent**, using the `Grant Admin Consent` link from the permissions screen.
```

File: docs/reference/connector/docs/connectors-sharepoint-online.asciidoc [hunk 3]
```diff
@@ -546,13 +546,13 @@ Select an expiration date. (At this expiration date, you will need to generate a
...
```
Comment (leemthompo) [inline docs/reference/connector/docs/connectors-sharepoint-online.asciidoc:557]:
```suggestion
NOTE: If the `Comma-separated list of sites` configuration is set to `*` or if a user enables the toggle button `Enumerate all sites`, the connector requires `Sites.Read.All` permission.

* **Grant admin consent**, using the `Grant Admin Consent` link from the permissions screen.
```

**Review by seanstory — APPROVED** [review]
With this, do we need to cover any extra steps to tell the customer how to "select" which sites should be covered in this scope?

**Comment by leemthompo** [comment]
> With this, do we need to cover any extra steps to tell the customer how to "select" which sites should be covered in this scope?

Merged to get the change in ASAP, but let me know your thoughts on this.

**Comment by seanstory** [comment]
😱 it's even worse than I'd thought. I figured that when you choose "sites.selected" in their UI, you'd then get some dropdown where you could multi-select sites to have it apply to. 
Nope. Have to use powershell or the graph API to specifically grant the new application ID access to specific sites. See: https://blog.dan-toft.dk/2022/12/sites-selected-permissions/

**Comment by seanstory** [comment]
So we're going to need docs similar to what we have here: https://www.elastic.co/guide/en/elasticsearch/reference/8.16/es-connectors-sharepoint-online.html#es-connectors-sharepoint-online-oauth-app-permissions but to cover how to work with Sites.selected.

Of course Crest failed to mention any of this 💀

**Comment by leemthompo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.17|[](https://github.com/elastic/elasticsearch/pull/117633)|
|✅|8.16|[](https://github.com/elastic/elasticsearch/pull/117634)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by leemthompo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/elasticsearch/pull/117635)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**